### ISSUE TITLE
Clean Up ReadMe Documentation

### ISSUE THREAD

**Issue by tylerkrupicka** [intuit/postcss-themed#24]
Right now the documentation in the ReadMe is a little hard to follow and could use some improvements. I think it would be helpful to:

- Break apart the `modern` and `legacy` generation into separate sections which explain the differences in how they work and how you use them.
- Create a legend at the top of the readme that links to individual configuration options or examples.
- Clean up headers and documentation text.

**Comment by anjaliguptaz** [comment]
@tylerkrupicka  Is this still valid and needs to be done ?
